packages/core/src/index.ts

```diff
@@ -204,18 +207,49 @@ export const createConversation = (config: Config): ConversationHandler => {
     if (!state.contextSent) {
       state = { ...state, contextSent: true };
     }
-    return fetch(config.botUrl, {
-      method: "POST",
-      headers: {
-        ...config.headers,
-        "content-type": "application/json",
-      },
-      body: JSON.stringify(bodyWithContext),
-    }).then((res) => res.json());
+    if (isUsingWebSockets()) {
+      socket.send(JSON.stringify(bodyWithContext));
+      return Promise.resolve({ isPending: true } as any);
+    } else {
+      return fetch(config.botUrl, {
+        method: "POST",
+        headers: {
+          ...config.headers,
+          "content-type": "application/json",
+        },
+        body: JSON.stringify(bodyWithContext),
+      }).then((res: any) => res.json());
+    }
+  };
+
+  const isUsingWebSockets = () => {
+    return config.botUrl && config.botUrl.indexOf("wss://") === 0;
   };
 
   let subscribers: Subscriber[] = [];
 
+  if (isUsingWebSockets()) {
+    // open websocket
+    socket = new WebSocket(config.botUrl);
+
+    socket.onerror = function(error: any) {
+      // TODO: Handle unexpected socket errors
+      // @peter, should we just raise this as an event on the widget for the consumer to handle?
+    };
+
+    socket.onclose = function() {
+      // TODO: Handle unexpected socket errors
+      // @peter, should we just raise this as an event on the widget for the consumer to handle?
+      // if we start seeing these we can implement an auto-reconnect in here...
+    };
+
+    socket.onmessage = function(e) {
+      if (typeof e.data === "string") {
+        messageResposeHandler(JSON.parse(e.data));
```

Comment:
`JSON.parse` should go in a try/catch as an extra layer of protection against runtime errors.